### PR TITLE
feat: add the product card directive

### PR DESCRIPTION
**Pull request by spencerwalker**
* Clean up product list styles
* Clean up product card styles
* Fix product card bugs

### CODE REVIEW

**Comment by crhistianr** [comment]
There's something weird about this PR. for instance app.constants.json shouldn't show up as a changed file. Did you pull recently?

**Comment by spencerwalker** [comment]
Yeah I dont think i was using "rebase" properly... so it is showing all the file changes from the upstream merges I believe.

File: src/app/common/templates/card.product.html [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+        <a ui-sref="productDetail({productid: productCard.product.ID})">
+            <img class="img-responsive c-product-card__img" ng-src="{{productCard.product.xp.image.URL || 'http://placehold.it/300x300'}}"/>
+        </a>
     </div>
+</div>
+<div class="c-product-card__body">
+    <!--Product Info-->
+    <!--<div class="text-right c-product-card__icons hidden-xs" oc-prevent-click>
+        <ordercloud-product-quick-view product="productCard.product" current-order="productCard.currentOrder"></ordercloud-product-quick-view>
+        <ordercloud-favorite-product product="productCard.product" current-user="productCard.currentUser"></ordercloud-favorite-product>
```
Comment (crhistianr) [inline src/app/common/templates/card.product.html]: Should this be here?

**Comment by robertsoniv** [comment]
I think his app.constants.json is right even though they look like they were changed

File: src/app/common/templates/card.product.html (hunk 1 above)
Comment (spencerwalker) [inline src/app/common/templates/card.product.html]: Uncommented that code. I will create separate subtasks for styling these.

**Comment by crhistianr** [comment]
js looks good to me. @robertsoniv want to check out his css?

**Pull request merged**